src/freedesktop.rs

```diff
@@ -1091,7 +1085,8 @@ mod tests {
     }
 }
 
-/// Converts a file system error to a crate `Error`
-fn fsys_err_to_unknown<P: AsRef<Path>>(path: P, orig: std::io::Error) -> Error {
-    Error::FileSystem { path: path.as_ref().to_owned(), kind: orig.kind() }
```

Comment:
This is the core of the PR let's keep this.

Reply:
But converting a tuple to Error via `from` looks prettier than using a function.